helpers/copy_tree.sh

```diff
@@ -140,29 +152,48 @@ if grep -q "%define multiple_rpms 1" "$modify_spec"; then
                 mv "$TREE_SPARSE"/etc/prop.default "$DEVICE"/system/etc/prop.default
             fi
         fi
-    elif [ "$TREE" = "vendor" ]; then
+    elif [ "$TREE" = "product" ] || [ "$TREE" = "vendor" ]; then
         if [ -f "$TREE_SPARSE"/build.prop ]; then
-            DEVICE=$(grep ro.vendor.product.name "$TREE_SPARSE"/build.prop | cut -d '_' -f2)
+            DEVICE=$(grep ro."$TREE".product.name "$TREE_SPARSE"/build.prop | cut -d '_' -f2)
             if [ -z "$DEVICE" ]; then
-                DEVICE=$(grep ro.product.vendor.name "$TREE_SPARSE"/build.prop | cut -d '_' -f2)
+                DEVICE=$(grep ro.product."$TREE".name "$TREE_SPARSE"/build.prop | cut -d '_' -f2)
             fi
             if [ -z "$DEVICE" ]; then
                 echo "Could not parse device name from $TREE_SPARSE/build.prop"
                 exit 1
             fi
-            mkdir -p "$DEVICE"/vendor
-            mv "$TREE_SPARSE"/build.prop "$DEVICE"/vendor
+            mkdir -p "$DEVICE"/"$TREE"
+            mv "$TREE_SPARSE"/build.prop "$DEVICE"/"$TREE"
 
             if [ -f "$TREE_SPARSE"/odm/etc/build.prop ]; then
-                mkdir -p "$DEVICE"/vendor/odm/etc
-                mv "$TREE_SPARSE"/odm/etc/build.prop "$DEVICE"/vendor/odm/etc
+                mkdir -p "$DEVICE"/"$TREE"/odm/etc
+                mv "$TREE_SPARSE"/odm/etc/build.prop "$DEVICE"/"$TREE"/odm/etc
             fi
         fi
 
         # Move Vendor Interface Object manifest to proper place
         if [ -f "$TREE_SPARSE"/etc/vintf/manifest.xml ]; then
-            mkdir -p "$DEVICE"/vendor/etc/vintf
-            mv "$TREE_SPARSE"/etc/vintf/manifest.xml "$DEVICE"/vendor/etc/vintf/manifest.xml
+            mkdir -p "$DEVICE"/"$TREE"/etc/vintf
+            mv "$TREE_SPARSE"/etc/vintf/manifest.xml "$DEVICE"/"$TREE"/etc/vintf/manifest.xml
+        fi
+    elif [ "$TREE" = "system_ext" ]; then
+        if [ -f "$TREE_SPARSE"/build.prop ]; then
+            DEVICE=$(grep ro."$TREE".product.name "$TREE_SPARSE"/build.prop | cut -d '_' -f3)
+            if [ -z "$DEVICE" ]; then
+                DEVICE=$(grep ro.product."$TREE".name "$TREE_SPARSE"/build.prop | cut -d '_' -f3)
+            fi
+            if [ -z "$DEVICE" ]; then
+                echo "Could not parse device name from $TREE_SPARSE/build.prop"
+                exit 1
+            fi
+            mkdir -p "$DEVICE"/"$TREE"
+            mv "$TREE_SPARSE"/build.prop "$DEVICE"/"$TREE"
+        fi
+
+        # Move Vendor Interface Object manifest to proper place
+        if [ -f "$TREE_SPARSE"/etc/vintf/manifest.xml ]; then
+            mkdir -p "$DEVICE"/"$TREE"/etc/vintf
+            mv "$TREE_SPARSE"/etc/vintf/manifest.xml "$DEVICE"/"$TREE"/etc/vintf/manifest.xml
         fi
     fi
```

Comment:
Please refactor the vintf bit, along the lines of
```suggestion
        fi
    elif [ "$TREE" = "system_ext" ]; then
        if [ -f "$TREE_SPARSE"/build.prop ]; then
            DEVICE=$(grep ro."$TREE".product.name "$TREE_SPARSE"/build.prop | cut -d '_' -f3)
            if [ -z "$DEVICE" ]; then
                DEVICE=$(grep ro.product."$TREE".name "$TREE_SPARSE"/build.prop | cut -d '_' -f3)
            fi
            if [ -z "$DEVICE" ]; then
                echo "Could not parse device name from $TREE_SPARSE/build.prop"
                exit 1
            fi
            mkdir -p "$DEVICE"/"$TREE"
            mv "$TREE_SPARSE"/build.prop "$DEVICE"/"$TREE"
        fi
    fi
    if [ "$TREE" = "product" ] || [ "$TREE" = "vendor" ] || [ "$TREE" = "system_ext" ]; then
        # Move Vendor Interface Object manifest to proper place
        if [ -f "$TREE_SPARSE"/etc/vintf/manifest.xml ]; then
                mkdir -p "$DEVICE"/"$TREE"/etc/vintf
            mv "$TREE_SPARSE"/etc/vintf/manifest.xml "$DEVICE"/"$TREE"/etc/vintf/manifest.xml
        fi
    fi
```

Reply:
Applied this to simplify code.